array-manipulation-3/src/flatten.js

```diff
@@ -0,0 +1,24 @@
+/* exported flatten */
+/*
+  - create storage for output
+  - look at each item one by one
+    - if the item's type is an object
+      - look at each item in that object
+        - add item to output
+    - otherwise
+      - add item to output
+*/
+
+function flatten(array) {
+  var flatArr = [];
+  for (var i = 0; i < array.length; i++) {
+    if (typeof array[i] === 'object') {
```

Comment:
I would use `Array.isArray()`, this could give a false positive for `object`s, which do not have a `length` property